apps/hubspot/src/locations/Page.tsx

```diff
@@ -1,16 +1,249 @@
+      setError(null);
+      try {
+        const cma = createClient(
+          { apiAdapter: sdk.cmaAdapter },
+          {
+            type: 'plain',
+            defaults: {
+              environmentId: sdk.ids.environment,
+              spaceId: sdk.ids.space,
+            },
+          }
+        );
+        const configService = new ConfigEntryService(cma, sdk.locales.default);
+        const connected = await configService.getConnectedFields();
+        setConnectedFields(connected);
+        setLocale(sdk.locales.default);
+        let displayField = 'title';
+        const entryIds = Object.keys(connected);
+        if (entryIds.length === 0) {
+          setEntries([]);
+          setDisplayFieldId('title');
+          setLoading(false);
+          return;
+        }
+        const fetchedEntries = [];
+        for (const entryId of entryIds) {
+          try {
+            const entry = await cma.entry.get({ entryId });
+            fetchedEntries.push(entry);
+            if (fetchedEntries.length === 1) {
+              const ct = await cma.contentType.get({ contentTypeId: entry.sys.contentType.sys.id });
+              displayField = ct.displayField || 'title';
+            }
+          } catch (e) {
+            // skip missing entry
+          }
+        }
+        setEntries(fetchedEntries);
+        setDisplayFieldId(displayField);
+      } catch (e) {
+        setEntries([]);
+        setError(
+          'The app cannot load content. Try refreshing, or reviewing your app configuration.'
+        );
+      } finally {
+        setLoading(false);
+      }
+    };
+    fetchData();
+  }, [sdk]);
+
+  const handleManageFields = (entry: any) => {
+    sdk.navigator.openEntry(entry.sys.id);
+  };
+
+  return (
+    <Flex justifyContent="center" paddingLeft="spacing2Xl" paddingRight="spacing2Xl">
+      <Box className={styles.container} padding="spacingL">
+        <Heading as="h1" marginBottom="spacingM">
+          Hubspot
+        </Heading>
+        <Text fontColor="gray600" marginBottom="spacingL">
+          View the details of your synced entry fields. Click Manage fields to connect or disconnect
+          content.
+        </Text>
+        {loading ? (
+          <EmptyState />
+        ) : error ? (
+          <DisplayMessage
+            title="The app cannot load content."
+            message="Try refreshing, or reviewing your app configuration"
+          />
+        ) : entries.length === 0 ? (
+          <DisplayMessage
+            title="No active Hubspot modules"
+            message="Once you have created modules, they will display here."
+          />
+        ) : (
+          <ConnectedEntriesTable
+            entries={entries}
+            connectedFields={connectedFields}
+            displayFieldId={displayFieldId}
+            locale={locale}
+            onManageFields={handleManageFields}
+          />
+        )}
+      </Box>
+    </Flex>
+  );
+};
+
+const ConnectedEntriesTable = ({
```

Comment:
💭 We could create a separate component for this! Since it seems to be similar to braze . We can leave this for the next page view PR